cmd/utils/auth.go

```diff
@@ -32,3 +49,82 @@ func GetCredentials() (*azidentity.ChainedTokenCredential, error) {
+// cachedInteractiveBrowserCredential is a credential that uses the interactive browser to authenticate and caches the token.
+// TODO: This is a workaround until the azidentity package supports caching, https://github.com/Azure/azure-sdk-for-go/issues/16643.
+type cachedInteractiveBrowserCredential struct {
+	client public.Client
+}
+
+func newCachedInteractiveBrowserCredential() (*cachedInteractiveBrowserCredential, error) {
+	file := path.Join(homeDir(), ".kubectl-az", "token-cache.json")
+	if err := os.MkdirAll(path.Dir(file), 0700); err != nil {
+		return nil, fmt.Errorf("creating cache directory: %w", err)
+	}
+
+	client, err := public.New(developerSignOnClientID,
+		public.WithCache(&tokenCache{file: file}),
+		public.WithAuthority(runtime.JoinPaths(string(azidentity.AzurePublicCloud), organizationsTenantID)),
+	)
+	if err != nil {
+		return nil, fmt.Errorf("creating public client: %w", err)
+	}
+
+	return &cachedInteractiveBrowserCredential{client: client}, nil
+}
+
+// GetToken implements the azcore.TokenCredential interface on cachedInteractiveBrowserCredential.
+func (c *cachedInteractiveBrowserCredential) GetToken(ctx context.Context, options policy.TokenRequestOptions) (*azcore.AccessToken, error) {
+	// TODO: may be this can be improved with https://github.com/Azure/kubectl-az/issues/11
+	var account public.Account
+	if len(c.client.Accounts()) > 0 {
+		account = c.client.Accounts()[len(c.client.Accounts())-1]
+	}
+	result, err := c.client.AcquireTokenSilent(ctx, options.Scopes, public.WithSilentAccount(account))
+	if err != nil {
+		result, err = c.client.AcquireTokenInteractive(ctx, options.Scopes)
+		if err != nil {
+			return nil, fmt.Errorf("acquiring interactive token: %w", err)
+		}
+	}
+	return &azcore.AccessToken{Token: result.AccessToken, ExpiresOn: result.ExpiresOn}, nil
+}
+
+// tokenCache implements basic file based cache.ExportReplace to be used with the public.Client.
+type tokenCache struct {
+	file string
+}
+
+func (t *tokenCache) Replace(cache cache.Unmarshaler, key string) {
+	data, err := os.ReadFile(t.file)
+	if err != nil && !errors.Is(err, fs.ErrNotExist) {
+		fmt.Fprintf(os.Stderr, "Warn: reading token cache: %s\n", err)
+	}
+	err = cache.Unmarshal(data)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Warn: unmarshaling token cache: %s\n", err)
+	}
+}
+
+func (t *tokenCache) Export(cache cache.Marshaler, key string) {
+	data, err := cache.Marshal()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Warn: marshaling token cache: %s\n", err)
+	}
+	var indentedData bytes.Buffer
+	if err = json.Indent(&indentedData, data, "", "  "); err != nil {
+		fmt.Fprintf(os.Stderr, "Warn: indenting token cache: %s\n", err)
+	}
+	err = os.WriteFile(t.file, indentedData.Bytes(), 0600)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Warn: writing token cache: %s\n", err)
+	}
+}
+
+func homeDir() string {
+	dir, err := os.UserHomeDir()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Warn: getting home directory: %s\n", err)
+	}
+	return dir
```

Comment:
It means we will create the `.kubectl-az` folder in the current directly if `UserHomeDir` fails. Could you please add a comment describing it?